src/dbally/prompts/few_shot.py

```diff
@@ -0,0 +1,37 @@
+import inspect
+import re
+import textwrap
+from typing import Callable, Union
+
+
+class FewShotExample:
+    """
+    A question:answer representation for few-shot prompting
+    """
+
+    def __init__(self, question: str, answer_expr: Union[str, Callable]) -> None:
+        """
+        Args:
+            question: sample question
+            answer_expr: it can be either a stringified expression or a lambda for greater safety and code completions.
+                in lambda case, it should be a single line expression (otherwise output correctness is not guaranteed)
+        """
+        self.question = question
+        self.answer_expr = answer_expr
+
+        if isinstance(self.answer_expr, str):
+            self.answer = self.answer_expr
```

Comment:
good point, although it depends a bit what we would like to fail first when the `answer_expr` cannot be parsed. if the logic is executed on `FewShotExample` creation and an exception happens we will immediately know that what we passed as a construction arguments is invalid, while having the logic in `cached_property` might give a false sense of no problems, because any exception would be thrown only when the property is accessed for the first time.
I'm fine with `cached_property` approach as it will certainly clean up initialization, so LMK if I should proceed and introduce `cached_property` :)

Reply:
We should leave that code in the `__init__` method to ensure that any syntax errors are caught as early as possible.

Reply:
Yeah, you're right, let's leave it as it is.